Cache management breaks shared installs of JBang in Linux

**Describe the bug**

In v0.103 a new cache management feature was included to validate the freshness of files downloaded by JBang. To check the latest download, it implemented `Files.setLastModifiedTime` among other validations. 

However, this implementation breaks shared installations of JBang as the Java method requires that the owner of the file is the only one allowed to change the file. Hence getting an `Operation Not Permitted` error.

**To Reproduce**
Steps to reproduce the behavior:
1. Create two users (user1,user2)
2. Assign the same $HOME to both users say `/home/user`
3. Install and setup the app (camel@apache/camel) using user1
4. Log in as user2 and try to run `camel init hello.js`

**Expected behavior**
As it worked in 0.102 the application succeeds in creating the file. However you will get the following error:

```
[jbang] [ERROR] Could not download https://github.com/apache/camel/blob/HEAD/jbang-catalog.json
[jbang] Run with --verbose for more details
```

**JBang version**
Cache: /home/user/.jbang/cache
Config: /home/user/.jbang
Repository:/home/user/.m2/repository
0.108.0

**Additional context**
References:
https://stackoverflow.com/questions/61011124/java-file-setlastmodified-fails-on-linux-when-you-are-not-the-file-owner

--- a/src/main/java/dev/jbang/source/ProjectBuilder.java
+++ b/src/main/java/dev/jbang/source/ProjectBuilder.java
@@ -221,7 +221,7 @@ public class ProjectBuilder {
 
 	private ResourceRef resolveChecked(ResourceResolver resolver, String resource) {
 		Util.verboseMsg("Resolving resource ref: " + resource);
-		boolean retryCandidate = catalogFile == null && !Util.isFresh() && Settings.getCacheEvict() != 0
+		boolean retryCandidate = catalogFile == null && !Util.isFresh() && Settings.getCacheEvict() > 0
 				&& (Catalog.isValidName(resource) || Catalog.isValidCatalogReference(resource)
 						|| Util.isRemoteRef(resource));
 		ResourceRef ref = null;

--- a/src/main/java/dev/jbang/util/Util.java
+++ b/src/main/java/dev/jbang/util/Util.java
@@ -1025,7 +1025,18 @@ public class Util {
 							verboseMsg(String.format("Not modified, using cached file %s for remote %s", cachedFile,
 									conn.getURL().toExternalForm()));
 							// Update cached file's last modified time
-							Files.setLastModifiedTime(cachedFile, FileTime.from(ZonedDateTime.now().toInstant()));
+							try {
+								Files.setLastModifiedTime(cachedFile, FileTime.from(ZonedDateTime.now().toInstant()));
+							} catch (IOException e) {
+								// There is an issue with Java not being able to set the file's last modified
+								// time
+								// on certain systems, resulting in an exception. There's not much we can do
+								// about
+								// that, so we'll just ignore it. It does mean that files affected by this will
+								// be
+								// re-downloaded every time.
+								verboseMsg("Unable to set last-modified time for " + cachedFile, e);
+							}
 							return cachedFile;
 						}
 					}

